extensions/ql-vscode/test/pure-tests/log-summary-parser.test.ts

```diff
@@ -0,0 +1,46 @@
+import { expect } from 'chai';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import 'mocha';
+
+import { parseVisualizerData } from '../../src/pure/log-summary-parser';
+
+describe('Evaluator log summary tests', async function () {
+  describe('for a valid summary text', async function () {
+    it('should return only valid EvaluatorLogData objects', async function () {
+      const validSummaryText = await fs.readFile(path.join(__dirname, 'evaluator-log-summaries/valid-summary.jsonl'), 'utf8');
+      const evaluatorLogData = parseVisualizerData(validSummaryText.toString());
+      expect (evaluatorLogData.length).to.eq(2);
+      for (const item of evaluatorLogData) {
+        expect(item.queryCausingWork).to.not.be.empty;
+        expect(item.predicateName).to.not.be.empty;
+        expect(item.millis).to.be.a('number');
+        expect(item.resultSize).to.be.a('number');
+        expect(item.ra).to.not.be.empty;
+        if (item.ra !== undefined) {
+          for (const pipeline of Object.entries(item.ra)) {
+            expect (pipeline).to.not.be.empty;
```

Comment:
Ah, you're right. I think I actually meant something like

```
for (const pipeline of item.ra) {
  expect (pipeline).to.not.be.empty;
}
```

because I wanted to check that each pipeline in the array (`item.ra`) was non-empty. But when I use the above code I get `TypeError: item.ra is not iterable` 🤔 which I find odd, because it is of type `Pipeline[]`.